src/multibase.js

```diff
@@ -0,0 +1,127 @@
+'use strict'
+
+const constants = require('./constants')
+const bs58 = require('bs58')
+
+exports = module.exports = multibase
+exports.encode = encode
+exports.decode = decode
+exports.isEncoded = isEncoded
+
+const errNotSupported = new Error('Unsupported encoding')
+
+// returns a new buffer with the multibase varint+code`
+function multibase (nameOrCode, buf) {
+  if (!buf) {
+    throw new Error('requires an encoded buffer')
+  }
+  const code = getCode(nameOrCode)
+  const codeBuf = new Buffer(code)
+
+  const name = getName(nameOrCode)
+  validEncode(name, buf)
+  return Buffer.concat([codeBuf, buf])
+}
+
+function encode (nameOrCode, buf) {
+  const name = getName(nameOrCode)
+
+  let encode
+
+  switch (name) {
+    case 'base58btc': {
+      encode = (buf) => { return new Buffer(bs58.encode(buf)) }
+    } break
+    default: throw errNotSupported
```

Comment:
How about using https://www.npmjs.com/package/base-x instead of bs58 (it is used by bs58)  and support the basic base encodings from the start?